### ISSUE TITLE
Suggestion: Add refcatoring for ArrayList

### ISSUE THREAD

**Issue by VahidN** [JosefPihrt/Roslynator#710]
Suppose we have an old code like this (belongs to .NET 1x):
```C#
ArrayList list = new ArrayList();
list.Add(1);
list.Add(2);
```
It would be really useful to have a refactoring for converting the `ArrayList` to a generic `List` to avoid the future boxings. The structure of this `List` should be detected from the usage of the `list` here: `List<int>`

**Comment by xtqqczze** [comment]
See also: [DE0006: Non-generic collections shouldn't be used](https://github.com/dotnet/platform-compat/blob/master/docs/DE0006.md)
